static_frame/core/index_hierarchy.py

```diff
@@ -2784,16 +2780,15 @@ def level_add(self: IH,
         index_cls = self._INDEX_CONSTRUCTOR if index_constructor is None else index_constructor._MUTABLE_CONSTRUCTOR # type: ignore
```

Comment:
This is old code, but it looks like we are always getting the `_MUTABLE_CONSTRUCTOR`; is that correct?

Reply:
I'm not sure what you mean... If `index_constructor is None`, then we get `self._INDEX_CONSTRUCTOR`, else, `index_constructor._MUTABLE_CONSTRUCTOR`

Regardless, this doesn't seem right. I think it should be:
```python
index_cls = self._INDEX_CONSTRUCTOR if index_constructor is None else index_constructor
```
Thoughts?

Reply:
Well, we are trying to do a few things at once here.

On review, I can see that `_INDEX_CONSTRUCTOR` is set to either GO or non-GO depending on if the class is `IndexHierarchyGO` or `IndexHierarchy`, respectively. So we do not need to vary that parameter if we are using it. But, if we get an `index_constructor` argument, we want to try to use the right variant. So I think what we want to do is this,

Some tests around this variability will help us see if we are covering all cases.

```python
if index_constructor is None:
    index_cls = self._INDEX_CONSTRUCTOR 
else: 
    index_cls = index_constructor._IMMUTABLE_CONSTRUCTOR if self.STATIC else index_constructor._MUTABLE_CONSTRUCTOR
```